graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlErrorHandler.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.graphql;
+
+import static java.util.Objects.requireNonNull;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+
+/**
+ * A handler that maps {@link ExecutionResult#getErrors()} or the specified {@link Throwable}
+ * to an {@link HttpResponse}.
+ */
+@UnstableApi
+@FunctionalInterface
+public interface GraphqlErrorHandler {
+
+    /**
+     * Return the {@link DefaultGraphqlErrorHandler}.
+     */
+    static GraphqlErrorHandler of() {
+        return DefaultGraphqlErrorHandler.INSTANCE;
+    }
+
+    /**
+     * Maps {@link ExecutionResult#getErrors()} or the specified {@link Throwable} to an {@link HttpResponse}.
+     */
+    @Nullable
+    HttpResponse handle(ServiceRequestContext ctx, ExecutionInput input,
+                        ExecutionResult result, MediaType negotiatedProduceType, @Nullable Throwable cause);
```

Comment:
Rather than passing in the `MediaType`, how about getting it from the `ServiceRequestContext`?

Reply:
Oh, we can get the value from `ServiceRequestContext` when we need it later! I was missing that point. Thank you for the good advice! I'll fix it right away. 😄

Reply:
@heowc I think we need to pass the `negotiatedProduceType` so that the implementation of this interface does not have to process getting the produce type again.
Also, we are going to add a setter for the default produce type. 
https://github.com/graphql/graphql-over-http/pull/215
(We can't just change the default type because it breaks the backward compatilblity.)
If so, we can't just call `GraphqlUtil.produceType` in the `handle` method.

Reply:
I am concerned that the argument value increases a lot, but since the `armeria-graphql` module is not yet mature, I think that it may need to be changed in the future. 🤔

Reply:
>  that the argument value increases a lot,

I don't get this. 😆 You mean the produce type will be changed a lot in the future?

Reply:
https://github.com/line/armeria/pull/4451
This is what I have expected that we needed. PTAL also. 😄

Reply:
Hi @heowc, @minwoox I'm sorry I'm late for personal reasons. I need your advice, because I'm still not good at making decisions

Reply:
Sorry about the late reply. 😉 
Let's keep that as is because this is an unstable API so we can add it later if we need the media type. 😉